tests/integration/test_orcid_import_legacy_tokens.py

```diff
@@ -0,0 +1,203 @@
+    'Josiah Carberry',
+)
+SAMPLE_USER_2 = (
+    '0000-0001-1234-1234',
+    '12345678-9abc-def1-2345-6789abcdef12',
+    'j.doe@orcid.org',
+    'John Doe',
+)
+SAMPLE_USER_EDITED = (
+    '0000-0002-1825-0097',
+    '00000000-0000-0000-0000-000000000000',
+    'j.carberry@orcid.org',
+    'Josiah Carberry',
+)
+
+
+def push_to_redis(user_record):
+    redis_url = current_app.config.get('CACHE_REDIS_URL')
+    r = StrictRedis.from_url(redis_url)
+    r.lpush('legacy_orcid_tokens', user_record)
+
+
+def flush_redis():
+    redis_url = current_app.config.get('CACHE_REDIS_URL')
+    r = StrictRedis.from_url(redis_url)
+    r.delete('legacy_orcid_tokens')
+
+
+def assert_db_has_n_legacy_records(n, record):
+    assert n == User.query.filter_by(email=record[2]).count()
+    assert n == RemoteAccount.query.filter_by(client_id=record[0]).count()
+    assert n == RemoteToken.query.filter_by(access_token=record[1]).count()
+    assert n == UserIdentity.query.filter_by(id=record[0]).count()
+
+
+def cleanup_record(record):
+    RemoteToken.query.filter_by(access_token=record[1]).delete()
+    RemoteAccount.query.filter_by(client_id=record[0]).delete()
+    UserIdentity.query.filter_by(id=record[0]).delete()
+    User.query.filter_by(email=record[2]).delete()
+
+
+def test_multiple_tuple_generator(app):
+    redis_url = current_app.config.get('CACHE_REDIS_URL')
+    r = StrictRedis.from_url(redis_url)
+    push_to_redis(SAMPLE_USER_2)
+    push_to_redis(SAMPLE_USER)
+
+    assert r.llen('legacy_orcid_tokens') == 2
+    gen = legacy_orcid_tuples()
+    assert list(gen) == [SAMPLE_USER, SAMPLE_USER_2]
+    assert r.llen('legacy_orcid_tokens') == 0
+    flush_redis()
+
+
+def test_import_multiple_orcid_tokens_no_user_exists(app):
+    """Create two users and all the associate entries."""
+    redis_url = current_app.config.get('CACHE_REDIS_URL')
+    r = StrictRedis.from_url(redis_url)
+    push_to_redis(SAMPLE_USER_2)
+    push_to_redis(SAMPLE_USER)
+
+    # Check initial state
+    assert r.llen('legacy_orcid_tokens') == 2
+    assert_db_has_n_legacy_records(0, SAMPLE_USER)
+    assert_db_has_n_legacy_records(0, SAMPLE_USER_2)
+
+    # Migrate
+    import_legacy_orcid_tokens()
+
+    # Check state after migration
+    assert not r.llen('legacy_orcid_tokens')
+    assert_db_has_n_legacy_records(1, SAMPLE_USER)
+    assert_db_has_n_legacy_records(1, SAMPLE_USER_2)
+
+    # Remove added records
+    cleanup_record(SAMPLE_USER)
+    cleanup_record(SAMPLE_USER_2)
+
+    # Check if properly deleted
+    assert_db_has_n_legacy_records(0, SAMPLE_USER)
+    assert_db_has_n_legacy_records(0, SAMPLE_USER_2)
+
+    flush_redis()
+
+
+def test_linked_user_with_token_exists(app):
+    """Ignore token, if already has one."""
+    assert_db_has_n_legacy_records(0, SAMPLE_USER)
+
+    # Register sample user
+    _register_user(orcid=SAMPLE_USER[0],
+                   token=SAMPLE_USER[1],
+                   email=SAMPLE_USER[2],
+                   name=SAMPLE_USER[3])
```

Comment:
You probably want to use a `dict` to store the user, and then use the double splat operator (`**`):
```python
SAMPLE_USER = {
    'orcid': ORCID,
    'token': TOKEN,
    [...]
}

[...]

_register_user(**SAMPLE_USER)
```
as this will actually call `_register_user` with the right keyword arguments.

Reply:
(The same applies for all the uses that follow!)